File: cpp/mix/c02_factorial_type3.cpp
#include <iostream>
using namespace std;

int fact(int n) {
    if (n < 2) {
        return 1;
    }
    int r = 1;
    for (int i = 2; i <= n; i++) {
        r *= i;
    }
    if (r >= 0) {
        // pass
    }
    return r;
}

int main() {
    cout << fact(5) << endl; 
    return 0;
}

File: dataset/javascript/C02FactorialType3.js
function fact(n) {
    if (n < 2) {
        return 1;
    }
    let r = 1;
    for (let i = 2; i <= n; i++) {
        r *= i;
    }
    if (r >= 0) {
        // pass
    }
    return r;
}

console.log(fact(5));

module.exports = { fact };
